### PR TITLE
Add NFT scale-testing helper; mitigate memory leaks in EET suites

### LINKED ISSUE #1604: Scale/performance testing for NFTs
**Summary**
We want to measure `secTransH`, etc. using states with various combinations of # of NFT types and instances.

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
- Closes #1604

**Notes for reviewer**:
- At the end of `HapiApiSpec.run()`, set the spec's infrastructure fields to `null` so these objects can be reclaimed.
    * This is important when running a series of EET clients that may each have very large registries.
- Add the `UniqueTokenStateSetup` client to create a state with a specified number of `NON_FUNGIBLE_UNIQUE` tokens and associated NFTs.
- Switch the values of the `SuiteRunner` `CATEGORY_MAP` from `HapiApiSuite` _instances_ to `Supplier<HapiApiSuite>`s so that the same suite can be repeated multiple times in JRS job without risk of memory leaks or corrupt spec internals.

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/suites/perf/token/UniqueTokenStateSetup.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+		List<HapiSpecOperation> ans = new ArrayList<>();
+		while (createdSoFar.get() < NUM_UNIQ_TOKENS) {
+			var thisBurst = Math.min(NUM_UNIQ_TOKENS - createdSoFar.get(), perBurst);
+			final var burst = inParallel(IntStream.range(0, thisBurst)
+					.mapToObj(i -> tokenCreate(uniqueTokenNameFn.apply(i + createdSoFar.get()))
+							.payingWith(GENESIS)
+							.tokenType(NON_FUNGIBLE_UNIQUE)
+							.deferStatusResolution()
+							.noLogging()
+							.supplyType(INFINITE)
```
Comment (ljianghedera) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/perf/token/UniqueTokenStateSetup.java:200]: nit: should this be of suplly type `FINITE`?

**Review by ljianghedera — APPROVED** [review]
LGTM.

**Review by qnswirlds — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Pull request merged**